src/node_perf.cc

```diff
@@ -373,6 +375,45 @@ void Timerify(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(wrap);
 }
 
+static void on_uv_interval(uv_timer_t* req) {
+  Environment* env = reinterpret_cast<Environment*>(req->data);
+
+  uint64_t time = uv_hrtime();
+  uint64_t delta = time - env->get_event_loop_delay_prev();
+
+  if (delta > EVENT_LOOP_DELAY_TIMER_THRESHOLD_NS) {
+    delta = (delta - EVENT_LOOP_DELAY_TIMER_RESOLUTION_NS) / 1e6;
+    if (delta > 0xffffffff) delta = 0xffffffff;
+    AliasedBuffer<uint32_t, Uint32Array>& buf = env->event_loop_delay();
+    buf[0] += (uint32_t)delta;
+    buf[1] = buf[1] + 1;  // The ++ operator is not defined
+    TRACE_COUNTER1(TRACING_CATEGORY_NODE2(perf, event_loop),
+                   "event loop delay",
+                   env->event_loop_delay()[0]);
+  }
+
+  env->set_event_loop_delay_prev(time);
+}
+
+void StartEventLoopTimer(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uv_timer_t* timer = env->event_loop_delay_timer();
+  uv_timer_init(uv_default_loop(), timer);
```

Comment:
Can you call this in `Environment::Start` instead, and add a call in `Environment::RegisterHandleCleanups()`? Otherwise I think we never explicitly close the timer handle.

Reply:
Hmm... calling in Environment::Start wouldn't make sense. The timer should not be started by default. Adding a handle cleanup call makes sense when the thing is running tho.

Reply:
@jasnell I’m only talking about `uv_timer_init()`, not `uv_timer_start()` (similar to the other handles in `Environment`)

Oh, and, just noticing, we don’t want to use `uv_default_loop()` :)

Reply:
heh... sigh... brain still getting back engaged after the holiday break :-)